include/seqan3/search/dream_index/interleaved_bloom_filter.hpp

```diff
@@ -0,0 +1,518 @@
+        using sdsl::bit_vector::end;
+        using sdsl::bit_vector::operator==;
+        using sdsl::bit_vector::operator[];
+
+    private:
+        friend class interleaved_bloom_filter;
+        using sdsl::bit_vector::resize;
+        using sdsl::bit_vector::set_int;
+    };
+
+private:
+    //!\cond
+    template <data_layout data_layout_mode>
+    friend class interleaved_bloom_filter;
+    //!\endcond
+
+    //!\brief The underlying datatype to use.
+    using data_type = std::conditional_t<data_layout_mode_ == data_layout::uncompressed,
+                                         sdsl::bit_vector,
+                                         sdsl::sd_vector<>>;
+
+    //!\brief The number of bins.
+    size_t bins_{};
+    //!\brief The size of each bin in bits.
+    size_t bin_size_{};
+    //!\brief The number of 64 bit integers used to represent the numerical value `bins`.
+    size_t bin_width{};
+    //!\brief How big is a block of `bin_width` many 64-bit integers.
+    size_t block_size{};
+    //!\brief How many blocks fit in the bitvector.
+    size_t block_count{};
+    //!\brief The number of hash functions.
+    size_t hash_funs{};
+    //!\brief The bitvector.
+    data_type data{};
+    //!\brief Precalculated values for hashing.
+    std::vector<size_t> pre_hash{};
+    //!\brief Shift value for hashing.
+    static constexpr size_t shift{27};
+    //!\brief Random seed for hashing.
+    static constexpr size_t seed{0x90b45d39fb6da1fa};
+    //!\brief The result buffer for a `bulk_contains()` query.
+    mutable binning_bitvector result_buffer{};
+
+    /*!\brief Perturbs a value and fits it into the vector.
+     * \param h The value to process.
+     */
+    constexpr void hash_and_fit(size_t & h) const
+    {
+        h ^= h >> shift; // Basically Fibonacci hashing
+        h %= block_count;
+        h *= block_size;
+    }
+
+public:
+    //!\brief Indicates whether the Interleaved Bloom Filter is compressed.
+    static constexpr data_layout data_layout_mode = data_layout_mode_;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    interleaved_bloom_filter() = default; //!< Defaulted.
+    //!\brief Resize `result_buffer` instead of copying.
+    interleaved_bloom_filter(interleaved_bloom_filter const & rhs)
+    {
+        std::tie(bins_, bin_size_, bin_width, block_size, block_count, data, pre_hash) =
+            std::tie(rhs.bins_, rhs.bin_size_, rhs.bin_width, rhs.block_size, rhs.block_count, rhs.data, rhs.pre_hash);
+        result_buffer.resize(bins_);
+    }
+    //!\brief Resize `result_buffer` instead of copying.
+    interleaved_bloom_filter & operator=(interleaved_bloom_filter const & rhs)
+    {
+        std::tie(bins_, bin_size_, bin_width, block_size, block_count, data, pre_hash) =
+            std::tie(rhs.bins_, rhs.bin_size_, rhs.bin_width, rhs.block_size, rhs.block_count, rhs.data, rhs.pre_hash);
+        result_buffer.resize(bins_);
+        return *this;
+    }
```

Comment:
Since we delete the copy ctor/assignment for the `binning_bitvector`, we have to customise both here.